src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,206 @@
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     */
+    function deregisterOperator(bytes32 operatorId, bool completeDeregistration, bytes calldata quorumNumbers, bytes32[] memory operatorIdsToSwap, uint32 globalOperatorListIndex) external onlyRegistryCoordinator {
+        require(quorumNumbers.length == operatorIdsToSwap.length, "IndexRegistry.deregisterOperator: quorumNumbers and operatorIdsToSwap must be the same length");
+
+        for (uint i = 0; i < quorumNumbers.length; i++) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            uint32 indexToRemove = operatorIdToIndexHistory[operatorId][quorumNumber][operatorIdToIndexHistory[operatorId][quorumNumber].length - 1].index;
+            _processOperatorRemoval(operatorId, quorumNumber, indexToRemove, operatorIdsToSwap[i]);
+            _updateTotalOperatorHistory(quorumNumber, totalOperatorsHistory[quorumNumber][totalOperatorsHistory[quorumNumber].length - 1].index - 1);
+        }
+
+        // remove operator from globalOperatorList if this is a complete deregistration
+        if(completeDeregistration){
+            _removeOperatorFromGlobalOperatorList(globalOperatorListIndex);  
+        }
+    }
+
+    /**
+     * @notice Looks up the `operator`'s index for `quorumNumber` at the specified `blockNumber` using the `index`.
+     * @param operatorId is the id of the operator for which the index is desired
+     * @param quorumNumber is the quorum number for which the operator index is desired
+     * @param blockNumber is the block number at which the index of the operator is desired
+     * @param index Used to specify the entry within the dynamic array `operatorIdToIndexHistory[operatorId]` to 
+     * read data from
+     * @dev Function will revert in the event that the specified `index` input does not identify the appropriate entry in the
+     * array `operatorIdToIndexHistory[operatorId][quorumNumber]` to pull the info from.
+     */
+    function getOperatorIndexForQuorumAtBlockNumberByIndex(bytes32 operatorId, uint8 quorumNumber, uint32 blockNumber, uint32 index) external view returns (uint32){
+        OperatorIndex memory operatorIndexToCheck = operatorIdToIndexHistory[operatorId][quorumNumber][index];
+
+        //blocknumber must be before the "index'th" entry's toBlockNumber
+        require(blockNumber <= operatorIndexToCheck.toBlockNumber, "IndexRegistry.getOperatorIndexForQuorumAtBlockNumberByIndex: provided index is too far in the future for provided block number");
+       
+        //if there is an index update before the "index'th" update, the blocknumber must be after than the previous entry's toBlockNumber
```

Comment:
typo: `after than` => `after`

Reply:
this typo is also present elsewhere. should find-and-replace.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/b03c92cfbb9bfd3366ad0bb1c11c20d3cedf29c1